integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/NewCarbonDataLoadRDD.scala

```diff
@@ -381,8 +381,10 @@ class NewDataFrameLoaderRDD[K, V](
         carbonLoadModel.getTableName + CarbonCommonConstants.UNDERSCORE + theSplit.index
       try {
 
-        loadMetadataDetails.setPartitionCount(CarbonTablePath.DEPRECATED_PATITION_ID)
-        loadMetadataDetails.setSegmentStatus(SegmentStatus.SUCCESS)
+        if (!carbonLoadModel.isCarbonNonTransactionalTable) {
```

Comment:
this check should not be present, only writing to tablestatus needs to be avoided. All the remaining flow should be same.

Reply:
Done